stairway/src/main/java/bio/terra/stairway/QueueInterface.java

```diff
@@ -0,0 +1,62 @@
+package bio.terra.stairway;
+
+public interface QueueInterface {
+
+  /**
+   * In a cluster configuration (e.g., Kubernetes), multiple service instances containing Stairway
+   * instances, cooperate the provide the service. In that configuration, Stairway uses a queue to
+   * share work across instances and to move work during shutdown. Stairway requires a transactional
+   * queue with at-least-once delivery semantics.
+   *
+   * <p>The queue interface is the abstraction Stairway uses to put messages onto the queue and read
+   * them from the queue for processing. At the time of this writing, the only message passed is the
+   * request for another Stairway instance to run a flight.
+   *
+   * <p>We need the queue abstraction to support multiple cloud platforms. On GCP, we use PubSub
+   * queues. On Azure, there are several messaging options.
```

Comment:
Thanks, these are great comments.
nit: move this part up as the class comment and leave the rest of this as the comment on `dispatchMessages`?